kernel/src/fs/file_table.rs

```diff
@@ -54,13 +52,12 @@ impl FileTable {
             let mode = InodeMode::S_IWUSR;
             fs_resolver.open(&tty_path, flags, mode.bits()).unwrap()
         };
-        table.put(FileTableEntry::new(Arc::new(stdin), FdFlags::empty()));
-        table.put(FileTableEntry::new(Arc::new(stdout), FdFlags::empty()));
-        table.put(FileTableEntry::new(Arc::new(stderr), FdFlags::empty()));
-        Self {
-            table,
-            subject: Subject::new(),
-        }
+        self.table
+            .put(FileTableEntry::new(Arc::new(stdin), FdFlags::empty()));
```

Comment:
Either the semantics or the implementation of this method is flawed. Before this method is called, the file table could have any number of file descriptors. So appending three entries to such a file table may not necessarily set the stdin, stdout, and stderr FDs.

I think you should remove `set_stdio` from `FileTable` and implement the same logic in the parent module's `init_in_first_process` function, which is the only caller of this function. The context of this caller guarantees that the file table is empty initially.